sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.fileconfig;
+
+import static java.util.stream.Collectors.toList;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.StructuredConfigProperties;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of {@link StructuredConfigProperties} which uses a file configuration model as a
+ * source.
+ *
+ * @see #getStructured(String) Accessing nested maps
+ * @see #getStructuredList(String) Accessing lists of maps
+ * @see FileConfiguration#toConfigProperties(Object) Converting configuration model to properties
+ */
+final class YamlStructuredConfigProperties implements StructuredConfigProperties {
+
+  /** Values are {@link #isPrimitive(Object)}, {@link List} of scalars. */
+  private final Map<String, Object> simpleEntries = new HashMap<>();
+
+  private final Map<String, List<StructuredConfigProperties>> listEntries = new HashMap<>();
+  private final Map<String, StructuredConfigProperties> mapEntries = new HashMap<>();
+
+  @SuppressWarnings("unchecked")
+  YamlStructuredConfigProperties(Map<String, Object> properties) {
```

Comment:
There's a lot of work being done in this constructor! Might prefer to keep the constructor simple and just assign fields...which means making this a static factory method, something like `YamlStructuredConfigProperties.fromMap(...)`.